lib/chef/resource/archive_file.rb

```diff
@@ -102,8 +102,11 @@ class ArchiveFile < Chef::Resource
         end
 
         if new_resource.owner || new_resource.group
-          converge_by("set owner of #{new_resource.destination} to #{new_resource.owner}:#{new_resource.group}") do
-            FileUtils.chown_R(new_resource.owner, new_resource.group, new_resource.destination)
+          converge_by("set owner of files extracted in #{new_resource.destination} to #{new_resource.owner}:#{new_resource.group}") do
+            archive = Archive::Reader.open_filename(new_resource.path)
+            archive.each_entry do |e|
+              FileUtils.chown_R(new_resource.owner, new_resource.group, "#{new_resource.destination}/#{e.pathname}")
```

Comment:
i think if each_entry includes directories then the `chown_R` is going to recursively chown files that might not be in the archive and will re-chown files that are in the archive which is probably not what you want.  OTOH if each_entry doesn't include directories then the chown_R isn't doing anything useful...

Reply:
Good point, I think I can arrange that. I'll give it another go sometime this week by constructing a full list of the archive contents and applying using plain `chown` to it.

Reply:
I played a bit with irb and confirmed `each_entry` returns the folders too